### PR TITLE
Now user cannot create data source with empty name (spaces)

### PR DESCRIPTION
**Pull request by MaksymShokin**
@tonytlwu @squallstar 

## Issue
Fliplet/fliplet-studio#3773

## Description
Changed confirm to Fliplet.Modal.confirm. Added alert when the user is trying to submit empty (only spaces) name.

## Screenshots/screencasts
edit empty ds

## Backward compatibility
This change is fully backward compatible.

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -560,21 +560,26 @@ $('#app')
...
-
-    if (!sourceName) {
-      return;
-    }
+    Fliplet.Modal.prompt({
+      title: 'New data source',
+      message: 'Enter a data source name',
+    }).then(function (result) {
+      var dataSourceName = result.trim();
+      if (!dataSourceName) {
```
Comment (tonytlwu) [inline js/interface.js]: The result could come back with `null` (cancelled) or `""`. They should be treated differently.

If it's cancelled, no further action should be taken, including any alerts.

If it comes back with an empty string, the alert defined here should be shown. After the alert is dismissed, the user should be shown the same prompt modal again so they can continue.

Comment (tonytlwu) [inline js/interface.js]: The result could come back with `null` (cancelled) or `""`. They should be treated differently.

If it's cancelled, no further action should be taken, including any alerts.

If it comes back with an empty string, the alert defined here should be shown. After the alert is dismissed, the user should be shown the same prompt modal again so they can continue.

Comment (MaksymShokin) [reply js/interface.js]: Fixed

File: js/interface.js [hunk 2]
```diff
@@ -360,6 +360,37 @@ function browseDataSource(id) {
     });
 }
 
+function createDataSource() {
+  Fliplet.Modal.prompt({
+  title: 'New data source',
```
Comment (squallstar) [inline js/interface.js]: indentation is incorrect

Comment (MaksymShokin) [reply js/interface.js]: Fixed

**Comment by squallstar** [comment]
@tonytlwu bump ^

File: js/interface.js [hunk 3]
```diff
@@ -360,6 +360,37 @@ function browseDataSource(id) {
...
 
+function createDataSource() {
+  Fliplet.Modal.prompt({
+    title: 'New data source',
+    message: 'Enter a data source name',
+  }).then(function (result) {
+    if (result === null) {
+      return;
+    }
+    var dataSourceName = result.trim();
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion

    var dataSourceName = result.trim();
```

File: js/interface.js [hunk 4]
```diff
@@ -360,6 +360,37 @@ function browseDataSource(id) {
...
+    if (!dataSourceName) {
```
Comment (tonytlwu) [inline js/interface.js]:
```suggestion

    if (!dataSourceName) {
```

**Pull request merged**